lib/policy_machine_storage_adapters/active_record/postgresql.rb

```diff
@@ -74,6 +74,40 @@ def self.operations_for_operation_sets(operation_set_ids, operation_names = nil)
         # ]
         connection.execute(sanitized_query)
       end
+
+      def self.accessible_objects_for_operations(user_id, operation_names, options)
+        field = options[:fields].first
+        filters = options.dig(:filters, :user_attributes) || {}
+
+        query = <<~SQL.squish
+          SELECT *
+          FROM pm_accessible_objects_for_operations(?, ?, ?, ?)
+        SQL
+
+        sanitized_query = sanitize_sql_for_assignment([
+          query,
+          user_id,
+          ActiveRecord.array_to_string(operation_names),
+          field,
+          JSON.dump(filters)
+        ])
+
+        # [
+        #   { 'unique_identifier' => 'op1', 'objects' => '{obj1,obj2,obj3}' },
+        #   { 'unique_identifier' => 'op2', 'objects' => '{obj1,obj2,obj3}' },
+        # ]
+        result = connection.execute(sanitized_query).to_a
+
+        # {
+        #    'op1' => [{ :field => 'obj1' }, { :field => 'obj2' }, { :field => 'obj3' }],
+        #    'op2' => [{ :field => 'obj2' }, { :field => 'obj3' }, { :field => 'obj4' }],
+        # }
+        result.map do |h|
+          objects = ActiveRecord.string_to_array(h['objects'])
+          value = objects.map { |o| { field => o }}
+          [h['unique_identifier'], value]
+        end.to_h
```

Comment:
i think `each_with_object` would be slightly more efficient here as it would only do one loop
```suggestion
        result.each_with_object({}) do |result_hash, output|
          objects = ActiveRecord.string_to_array(result_hash['objects'])
          key = result_hash['unique_identifier']
          output[key] = objects.map { |o| { field => o }}
        end
```

Reply:
24b4a9d